frontend/components/text-classifier/header/TextClassificationHeader.vue

```diff
@@ -116,27 +123,25 @@ export default {
       });
     },
     async onValidate(records) {
-      const filterdRecord = records.filter(
-        (r) => r.annotation || r.predicted_as || r.multi_label
+      const filteredRecord = records.filter(
+        (r) => r.currentAnnotation || r.predicted_as || r.multi_label
       );
+      const validatedRecords = filteredRecord.map((record) => {
+        const annotationLabels = record.currentAnnotation?.labels;
+        const modelPredictionLabels = this.formatLabels(record.predicted_as);
+        return {
+          ...record,
+          currentAnnotation: null,
+          annotation: {
+            labels: annotationLabels || modelPredictionLabels || [],
+          },
```

Comment:
1/ use another variable for the condition annotationLabels || modelPredictionLabels || [],
2/ use a factory method to create the return object